### ISSUE TITLE
Add FLAC sound file support

### ISSUE THREAD

**Issue by Kefta** [Facepunch/garrysmod-requests#1331]
[FLAC](https://xiph.org/flac/) is a free, lossless audio format that requires no licensing to use the codec. It's the most space-efficient way (other than Apple's propriety ALAC) to store and play lossless audio. Adoption has been wider every year, and would be a great alternative to GMod's current support for MP3 (lossy), WAV (lossless but uncompressed), and OGG (can be lossy or lossless, but is simply a container format for other audio files).

**Comment by Kefta** [comment]
BASS has addons available on https://www.un4seen.com/ for both FLAC and ALAC file playback.

**Comment by thegrb93** [comment]
FLAC is heavy on the cpu, isn't it?

**Comment by Kefta** [comment]
Not to my knowledge.

**Comment by thegrb93** [comment]
Maybe I was thinking of a different one. Tried out FLAC and seems fine.

**Comment by FarukGamer** [comment]
Someone done a comparison of the different audio codecs  and their Resource usage, well.. This comparison is related to phones, but it could be imagined how the usage be like.

https://hydrogenaud.io/index.php/topic,92235.0.html

**Comment by Kefta** [comment]
Yeah I would have expected OGG to be the greatest since it is a container for other formats.
